pkg/cmd/kafka/topic/consume/consume.go

```diff
@@ -0,0 +1,182 @@
+package consume
+
+import (
+	kafkaflagutil "github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/flagutil"
+
+	"strings"
+
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/kafkacmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/dump"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/contextutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
+	"github.com/spf13/cobra"
+)
+
+type options struct {
+	topicName string
+	kafkaID   string
+	partition int32
+	timestamp string
+	limit     int32
+	offset    int64
+	wait      bool
+	outputFormat string
+
+	f *factory.Factory
+}
+
+// row is the details of a record produced needed to print to a table
+type kafkaRow struct {
+	Topic     string `json:"topic" header:"Topic"`
+	Key       string `json:"key" header:"Key"`
+	Value     string `json:"value" header:"Value"`
+	Partition int32  `json:"partition" header:"Partition"`
+	Offset 	  int64  `json:"offset" header:"Offset"`
+}
+
+// NewComsumeTopicCommand creates a new command for producing to a kafka topic.
+func NewConsumeTopicCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "consume",
+		Short:   "consume short",
+		Long:    "consume long",
+		Example: "consume example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if opts.kafkaID == "" {
+
+				kafkaInstance, err := contextutil.GetCurrentKafkaInstance(f)
+				if err != nil {
+					return err
+				}
+
+				opts.kafkaID = kafkaInstance.GetId()
+			}
+
+			return runCmd(opts)
+		},
+	}
+
+	flags := kafkaflagutil.NewFlagSet(cmd, f.Localizer)
+
+	flags.AddOutput(&opts.outputFormat)
+	flags.StringVar(&opts.topicName, "name", "", f.Localizer.MustLocalize("kafka.topic.common.flag.name.description"))
+	flags.Int32Var(&opts.partition, "partition", 0, f.Localizer.MustLocalize("kafka.topic.consume.flag.partition.description"))
+	flags.StringVar(&opts.timestamp, "timestamp", "", f.Localizer.MustLocalize("kafka.topic.consume.flag.timestamp.description"))
```

Comment:
Do we need date flag as well?

Reply:
There are number of date formats. We need to be specific what timestamp is. Databases have sepecific format associated with timestamp etc.

More: https://en.wikipedia.org/wiki/ISO_8601

Reply:
Oh okay I understand, just something to specify the format used